src/world/partner/sushie.c

```diff
@@ -97,15 +205,127 @@ s32 func_802BF5A0_320310(Evt* evt, s32 arg1) {
             }
             break;
     }
-    return 0;
+
+    return ApiStatus_BLOCK;
+}
+
+void func_802BF920_320690(Npc* npc) {
+    if (D_8010C954 != 0) {
+        D_8010C954 = NULL;
+        npc->flags = D_802BFDF8_320B68->unk_08;
+        D_802BFDF8_320B68->unk_04 = 0;
+        partner_clear_player_tracking(npc);
+    }
 }
 
-INCLUDE_ASM(s32, "world/partner/sushie", func_802BF920_320690);
+s32 func_802BF964_3206D4(Evt* script, s32 isInitialCall) {
+    Npc* npc = script->owner2.npc;
 
-INCLUDE_ASM(s32, "world/partner/sushie", func_802BF964_3206D4);
+    if (isInitialCall != 0) {
+        partner_init_put_away(npc);
+        gPlayerStatusPtr->animFlags &= 0xFFBFFFFF;
```

Comment:
```suggestion
        gPlayerStatusPtr->animFlags &= ~0x00400000;
```

Reply:
Also take a look at enums.h and see if there are flags defined for this - If there isn't, then please setup a new enum for AnimationFlags